src/Microsoft.SqlTools.Authentication/Authenticator.cs

```diff
@@ -0,0 +1,242 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+
+namespace Microsoft.SqlTools.Authentication
+{
+    /// <summary>
+    /// Provides APIs to acquire access token using MSAL.NET v4 with provided <see cref="AuthenticationParams"/>.
+    /// </summary>
+    public class Authenticator
+    {
+        private const string ApplicationClientId = "a69788c6-1d43-44ed-9ca3-b83e194da255";
+        private const string ApplicationName = "azuredatastudio";
```

Comment:
this library will be used in other applications, such as vscode or potentially other CLIs (mssql-scripter, etc., though this is a less important use-case).  Should this be configurable?

Reply:
Agreed, I can accept this in constructor, I was also thinking of extracting full cache location out of authenticator if so it can be controlled by caller, and as we're looking at other tools, I think it makes most sense to do so..